### DISCUSSION TITLE
Undefined array key "control_2vias"

### DISCUSSION THREAD

**Discussion by wanderleyrodrigo** [RamonSilva20/mapos#2119] [General]
Boa tarde, 
Fiz o update do sistema para a versão mais recente com PHP8.1 e o mesmo está ocorrendo o erro da imagem abaixo.
![image](https://user-images.githubusercontent.com/122235964/236907889-fb4ec956-cd72-4e5e-a931-168224006951.png)
O que seria possível fazer para resolver o problema.
Ocorre quando vou imprimir a OS.
Atenciosamente, 
Wanderley Rodrigo de Souza

**Comment by willph** [comment]
Olá tudo bom?

Ja conseguiu resolver seu problema?

**Comment by wanderleyrodrigo** [comment]
Olá boa tarde, não consegui.

Achei muito estranho, pois quando instalado ele do zero roda, mas quando
importo o backup para ter acesso aos meu dados de OS para não perder não
funciona começa o erro

Atenciosamente,

Em seg, 15 de mai de 2023 15:58, Wilmerson ***@***.***>
escreveu:

**Comment by willph** [comment]
Quando puder fazer um teste em outra instalação.

Importa e depois vai em configurações>sistema na aba atualizações e da um
clique em atualiza banco de dados.

E depois me diz o resultado.


Obrigado.

Em seg, 15 de mai de 2023 16:24, Professor Wanderley Rodrigo <
***@***.***> escreveu:

**Comment by wanderleyrodrigo** [comment]
Já fiz, ele após importar os dados não deixa eu nem abrir a aba
configurações, que já dá o erro.

Em seg, 15 de mai de 2023 16:30, Wilmerson ***@***.***>
escreveu:

**Comment by willph** [comment]
Tentou pelo script que fica na pasta updates no raiz do MapOs?

Lá tem os comandos SQL para atualização também.

Em seg, 15 de mai de 2023 16:40, Professor Wanderley Rodrigo <
***@***.***> escreveu:

**Comment by wanderleyrodrigo** [comment]
Este não, vou tentar e te aviso.

Agradecido,

Em seg, 15 de mai de 2023 16:55, Wilmerson ***@***.***>
escreveu:

**Reply by depatriservicos** [reply]
alguma solução ? mesmo problema aqui ....

**Comment by willph** [comment]
Usa o arquivo SQL que está na pasta updates.

Só analisar a versão correspondente e atualiza.

Em sex, 19 de mai de 2023 13:20, depatriservicos ***@***.***>
escreveu:

**Comment by mingccm1** [comment]
Boa tarde, tive esse mesmo problema depois de atualizar a versão 4.39 para 4.40. Fiz a instalação do zero, com Xampp V 3.3.0 e PHP 8.1.17 e phpMyAdmin 5.2.1. Depois de importar o backup do SQL da versão 4.39 e fazer o update manual pela pasta updates, continua o mesmo erro, quando clico na aba sistema já acontece o erro reportado acima, não permite que clique no botão atualizar banco de dados. Parece que também não está salvando as modificações, quando acrescento desconto numa OS, não aparece esse desconto, mesmo confirmando que foi atualizado com sucesso. Obrigado

**Reply by mingccm1** [reply]
Boa Tarde a todos, resolvi o problema da tela control_2vias, executando esse comando (conforme a instrução dada no Issues). 
INSERT INTO configuracoes (idConfig, config, valor) VALUES (15, 'control_2vias', 0);
Deu certo e não aparece mais esse erro. Mas continuo com problema nos descontos, na tela fala que foi atualizado com sucesso, mas quando abro de novo, não foi aplicado o desconto. alguém com esse erro ?

**Comment by willph** [comment]
Faltou executar o restante do SQL que fica na pasta updates no raiz do
MapOs.

....4.39.0_to_4.40.0.... O arquivo está escrito.

Fica atento às linhas que já executou.

Se não tiver executado nenhuma, subir todo o arquivo.

Em ter, 23 de mai de 2023 14:22, mingccm1 ***@***.***>
escreveu:
